src/Datasource/QueryInterface.php

```diff
@@ -279,6 +279,7 @@ public function toArray(): array;
      *
      * @param \Cake\Datasource\RepositoryInterface $repository The default repository object to use
      * @return $this
+     * @deprecated
```

Comment:
The implementing element contains the deprecation message with get/set info.
So maybe we dont need to repeat the details here.